Compatibility with PHP 8.0

eclipxe13 opened the pull request:
PHPUnit is upgraded since this library supports PHP >= 7.2 and PHPUnit 8 supports from 7.2 until 8.0

- Fix XSLTProcessor::transformToDoc signature to match with PHP XSLTProcessor.
- Fix Math functions by casting to floats.
- Update tests to work with PHPUnit 8.
- Travis-CI: Fixed compatibility with XDebug 3.0.
- Travis-CI: Due compatibility issues with PHP 8.0, do not install or test php-cs-fixer and phpstan.

**Important:** One pending task (that requires a different PR) is to upgrade PHPStan from version 0.11 to version 0.12.

eclipxe13 committed:
Upgrade dev dependencies to be installable using php >= 7.2

eclipxe13 committed:
Upgrade to PHPUnit 8 signatures

eclipxe13 committed:
Update to \XSLTProcessor::transformToDoc signature

eclipxe13 committed:
Modify deprecated assertContains to assertStringContainsString

eclipxe13 committed:
Modify deprecated assertNotContains to assertStringNotContainsString

eclipxe13 committed:
Cast to float since php math functions requires it

eclipxe13 committed:
Update travis to include PHP 8.0 on matrix

Remove friendsofphp/php-cs-fixer as dependency on PHP 8.0

eclipxe13 committed:
Fix travis build phpunit with xdebug 3 and code coverage

eclipxe13 committed:
Revert PHPStan version upgrade

eclipxe13 committed:
Remove dev tool PHPStan on PHP 8.0

frederikbosch commented:
Nice! We could drop 7.2 and only support 7.3/7.4/8.0 now. What do you think?

eclipxe13 commented:
It's up to you, PHP 7.2 has reached EOL since 2020-11-30.
If PHP 7.2 is drop then we can upgrade PHPUnit to version 9.

frederikbosch commented:
Let's go for PHP7.3+ only. I rather not support version that reached EOL.

eclipxe13 commented:
Just try to make it work upgrading to PHPUnit 9, there are two files that require to change on the test suite:

* `assertRegExp` is deprecated, substitute with `assertMatchesRegularExpression`
* `expectException(Error::class)` should be replaced with `expectError()`, also `expectExceptionMessage()` replace to `expectErrorMessage()`

The main issue I see on upgrade to PHPUnit 9 is that it has incompatible dependencies with PHPStan 0.11.
Also PHPStan 0.11 has been deprecated since 2019-10-22,

So, I would recommend to you follow the path:

1. Merge this PR that allows this library to run using PHP 8.0
2. Upgrade to PHPStan 0.12 and solve the all the issues (this is not trivial, level 6 is tedious)
3. Upgrade to PHPUnit 9 and PHP >= 7.3

PHPStan 0.12 counts 175 errors on max level:

- Level 2: 3 trivial errors
- Level 4: +8 errors, trivial but requires your review
- Level 6: +91 errors, need a lot of annotations on traversables and arrays
- Level 7: +8 errors
- Level 8: +65 errors

I can work with you on this process, what is your opinion?

frederikbosch commented:
Let's include the drop of PHP 7.2 in this PR by just removing it from the testsuite. Then I'll merge. I agree with you on the rest of path.

frederikbosch commented:
I can do the PHPStan upgrade to 0.12 no problem. I have done it before for my [mail package](https://github.com/genkgo/mail). Mostly it is improving docblocks with generics syntax.

eclipxe13 committed:
Drop PHP 7.2 support (EOL version)

eclipxe13 committed:
Improve Travis-CI build matrix

For each version the build runs only 2 times, not 3 as before.
Remove PHP 8.0 from allow failures section

eclipxe13 committed:
Upgrade php-cs-fixer (dev-dep) to fix build with lowest deps

frederikbosch commented:
Fantastic, great work!

vkondratjevs commented:
You can crate tag to php8 ?

frederikbosch commented:
Yes, done!
